### DISCUSSION TITLE
[help] compatibility with openxlsx?

### DISCUSSION THREAD

**Discussion by oTheAnalyst** [ropensci/targets#1290] [Help]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I have some code I'm trying to implement into a targets pipeline that is supposed to output an xlsx document at the end of the pipeline, the goal is to have the document connected to tableau. I'm  having issues trying to implement the code into my pipeline, I'm wondering if targets is at all compatible with openxlsx ? 

here is a snippet of the code below. The first function build list of rds files stored from targets, this was implemented as workaround to being able to use tar_objects() in the pipeline. 
``` r 
library(targets)
library(tidyverse)
library(openxlsx)
# the problem with code snippet is
# that I can't include tar_read in the target list.
# consider a work around
build_list <- function(paths) {
  file_paths <- as.data.frame.list(list.files(
    path = paths,
    pattern = "\\_data$",
    full.names = TRUE
  ))
  data_list <- readRDS(file_paths)
  return(data_list)
}

build_wb <- function(data_list) {
  sheets_list <- paste0("sheet_", seq(1, length(data_list)))
  wb <- createWorkbook()
  sheets_list %>%
    walk(~ addWorksheet(wb, sheetName = .))
  walk2(
    .x = sheets_list,
    .y = data_list,
    ~ writeData(wb,
      sheet = .x,
      x = .y
    )
  )
  saveWorkbook(wb,
    "tableau_upload.xlsx",
    overwrite = TRUE
  )
}
```

This more a question about I general approach I can use when trying to create an excel file using openxlsx and targets. If you have anymore refined approaches to creating an xlsx document using targets please let me know :)

**Comment by wlandau** [comment]
This is straightforward to do in `targets`. `tar_objects()` is not necessary because `openxlsx` can work with in-memory objects and does not need direct access to RDS files. Those in-memory objects are automatically saved by `targets` and can be used by downstream steps in the pipeline, as demonstrated in https://books.ropensci.org/targets/walkthrough.html.

Here is a complete `_targets.R` file which establishes a dataset and saves it in a workbook using `openxlsx`. After you run `tar_make()` to run the pipeline, you can open the output file `book.xlsx` in Excel to check that the data made it into a worksheet.

```r
# _targets.R file
library(targets)
tar_option_set(packages = "openxlsx")

save_workbook <- function(data, path) {
  book <- createWorkbook()
  addWorksheet(book, "cars")
  writeData(wb = book, sheet = "cars", x = data)
  saveWorkbook(wb = book, file = path, overwrite = TRUE)
  path
}

list(
  tar_target(data, mtcars),
  tar_target(book, save_workbook(data, "book.xlsx"), format = "file")
)
```

**Comment by oTheAnalyst** [comment]
thank you this helped! your explanation was clear, sorry I forgot to close this issue earlier.
